lib/ts/recipe/oauth2/recipeImplementation.ts

```diff
@@ -30,6 +30,37 @@ export default function getRecipeInterface(
     _appInfo: NormalisedAppinfo
 ): RecipeInterface {
     return {
+        getOAuth2Clients: async function (input, userContext) {
+            let response = await querier.sendGetRequest(
+                new NormalisedURLPath(`/recipe/oauth2/admin/clients`),
+                transformObjectKeys(input, "snake-case"),
+                userContext
+            );
+
+            if (response.status === "OK") {
+                let nextPageToken: string | undefined;
+                const linkHeader = response.headers.get("link");
+
+                const nextLinkMatch = linkHeader.match(/<([^>]+)>;\s*rel="next"/);
```

Comment:
Please add a short explanation comment here. An example of the header contents would be nice.

Reply:
done